src/confluent_kafka/src/Metadata.c

```diff
@@ -208,10 +208,18 @@ static PyObject *c_broker_to_py(Handle *self, PyObject *BrokerMetadata_type,
         }
         Py_DECREF(key);
 
-        if (cfl_PyObject_SetString(broker, "host",
-                                        c_broker.host) == -1) {
-                Py_DECREF(broker);
-                return NULL;
+        if (c_broker.host) {
+                if (cfl_PyObject_SetString(broker, "host",
+                                                c_broker.host) == -1) {
+                        Py_DECREF(broker);
+                        return NULL;
+                }
+        } else {
+                if (cfl_PyObject_SetItemString(broker, "host",
+                                                Py_None) == -1) {
+                        Py_DECREF(broker);
+                        return NULL;
+                }
```

Comment:
Please revert this change:

1. This function is used in deprecated API and will be removed in future versions
2. This change is not related to describe_topic
3. We can't reproduce this
4. It shouldn't be null as per the protocol

Reply:
Are you sure about that? Ultimately this function is called by `list_topics` in Metadata.c which sounds like it may be related to the segfaults I reported.

Were you able to isolate those segfaults to the one bit of code that you do want to change, in confluent_kafka.c?

Reply:
The host comes from the Metadata response in `list_topics` and is not nullable there
https://github.com/apache/kafka/blob/e8cd661becb08279702f1670bb8f1d816537adea/clients/src/main/resources/common/message/MetadataResponse.json#L54

otherwise it can come from list_group (deprecated) that in librdkafka only does a list + describe and in that case it's taken from the host of the coordinator that returned the DescribeGroup response here:
https://github.com/confluentinc/librdkafka/blob/c95d2ee950bda779c63843d376aab1d90ed8dfa9/src/rdkafka.c#L4739